app/shared/user-agent-client.js

```diff
@@ -17,10 +17,15 @@ import { logger } from './logging';
 import { UserAgentHttpClient } from './user-agent-http-client';
 
 class UserAgentClient extends EventEmitter {
-  constructor() {
+  // You'd think { initialConnectDelayMillis = 50, maxConnectDelayMillis = 10000 } would work.  In
+  // fact, you'd be crying in your beer, just like me: it doesn't when you call simply `new
+  // UserAgentClient()`.
+  constructor(options) {
```

Comment:
I believe `{initialConnectDelayMillis = 50, maxConnectDelayMillis = 10000}` does work, but it doesn't have any default param to assign to when there's no param available. Simply adding ` = {}` at the end will work. This comment is unproductive.